webapp/src/main/java/com/box/l10n/mojito/service/ai/translation/AITranslateJob.java

```diff
@@ -0,0 +1,257 @@
+package com.box.l10n.mojito.service.ai.translation;
+
+import com.box.l10n.mojito.JSR310Migration;
+import com.box.l10n.mojito.entity.Locale;
+import com.box.l10n.mojito.entity.Repository;
+import com.box.l10n.mojito.entity.RepositoryLocale;
+import com.box.l10n.mojito.entity.RepositoryLocaleAIPrompt;
+import com.box.l10n.mojito.entity.TMTextUnit;
+import com.box.l10n.mojito.entity.TMTextUnitVariant;
+import com.box.l10n.mojito.entity.TmTextUnitPendingMT;
+import com.box.l10n.mojito.quartz.QuartzPollableJob;
+import com.box.l10n.mojito.service.ai.LLMService;
+import com.box.l10n.mojito.service.ai.RepositoryLocaleAIPromptRepository;
+import com.box.l10n.mojito.service.repository.RepositoryRepository;
+import com.box.l10n.mojito.service.tm.TMService;
+import com.box.l10n.mojito.service.tm.TMTextUnitCurrentVariantRepository;
+import com.box.l10n.mojito.service.tm.TMTextUnitRepository;
+import io.micrometer.core.annotation.Timed;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tags;
+import java.time.Duration;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+
+@Component
+@ConditionalOnProperty(value = "l10n.ai.translation.enabled", havingValue = "true")
+public class AITranslateJob extends QuartzPollableJob<AITranslateJobInput, Void> {
+
+  static Logger logger = LoggerFactory.getLogger(AITranslateJob.class);
+
+  private static final String REPOSITORY_DEFAULT_PROMPT = "repository_default_prompt";
+
+  @Autowired TmTextUnitPendingMTRepository tmTextUnitPendingMTRepository;
+
+  @Autowired TMTextUnitRepository tmTextUnitRepository;
+
+  @Autowired TMTextUnitCurrentVariantRepository tmTextUnitCurrentVariantRepository;
+
+  @Autowired TMService tmService;
+
+  @Autowired LLMService llmService;
+
+  @Autowired MeterRegistry meterRegistry;
+
+  @Autowired RepositoryRepository repositoryRepository;
+
+  @Autowired RepositoryLocaleAIPromptRepository repositoryLocaleAIPromptRepository;
+
+  @Autowired AITranslationTextUnitFilterService aiTranslationTextUnitFilterService;
+
+  @Value("${l10n.ai.translation.pendingMT.expiryDuration:PT3H}")
+  Duration expiryDuration;
+
+  @Value("${l10n.ai.translation.reuseSourceOnLanguageMatch:false}")
+  boolean isReuseSourceOnLanguageMatch;
```

Comment:
Could we document here what this setting does in practice, with an example?

I don't quite recall the feature that this supports. An example I have in my head is this isn't a safe setting to use across all locales, e.g.: it might be safe to use for src:fr-FR & tgt:fr-CA but not safe for sr-Cyrl & sr-Latn or zh-CN & zh-TW.